datastream-kafka-connector/src/main/java/com/linkedin/datastream/connectors/kafka/AbstractKafkaConnector.java

```diff
@@ -118,6 +127,10 @@ public AbstractKafkaConnector(String connectorName, Properties config, GroupIdCo
     _clusterName = clusterName;
     _config = new KafkaBasedConnectorConfig(config);
     _groupIdConstructor = groupIdConstructor;
+    _dynamicMetricsManager = DynamicMetricsManager.getInstance();
+    _metricsPrefix = StringUtils.isBlank(connectorName) ? this.getClass().getSimpleName()
+        : connectorName + "." + this.getClass().getSimpleName();
+    _dynamicMetricsManager.registerGauge(_metricsPrefix, NUM_TASK_RESTARTS, () -> _numTaskRestarts);
```

Comment:
`_numTaskRestarts` will be accessed by different threads (for reading and writing). I suggest marking it `volatile`.